sql/ha_partition.cc

```diff
@@ -9102,14 +9131,40 @@ int ha_partition::loop_extra(enum ha_extra_function operation)
       In this case calling 'extra' can crash.
     */
     if (bitmap_is_set(&m_opened_partitions, i) &&
-        (tmp= m_file[i]->extra(operation)))
+        (tmp= f(m_file[i], param)))
       result= tmp;
   }
   /* Add all used partitions to be called in reset(). */
   bitmap_union(&m_partitions_to_reset, &m_part_info->lock_partitions);
   DBUG_RETURN(result);
 }
 
+static int extra_cb(handler *h, void *operation)
+{
+  return h->extra(*(enum ha_extra_function*)operation);
+}
+
+/*
+  Call extra on all partitions
+
+  SYNOPSIS
+    loop_extra()
+    operation             extra operation type
+
+  RETURN VALUE
+    >0                    Error code
+    0                     Success
+*/
+
+int ha_partition::loop_extra(enum ha_extra_function operation)
+{
+  int result;
+  DBUG_ENTER("ha_partition::loop_extra()");
+
+  result= loop_partitions(extra_cb, &operation);
+  DBUG_RETURN(result);
+}
```

Comment:
`loop_extra()` is empty stub. Use `loop_partitions()` everywhere.

Reply:
`loop_extra = loop_partitions(extra_cb, .)`
Using `loop_partitions` directly would increase patch significantly. Let's assume it as a shortcut.
Maybe I'd better just remove debug frame and make it inline?

Reply:
I looked at how much there `loop_extra()` references before writing the recommendation. It's not much.